src/core/toga/widgets/table.py

```diff
@@ -24,136 +19,129 @@ def __repr__(self):
-        if self.interface:
-            self.interface._impl.remove_node(node)
 
+    def item(self, row: int, column: int):
+        if isinstance(row and column, int):
+            return self.data[row].data[column]
+
+    def row(self, row: int) -> TableRow:
+        if row >= 0:
+            return self.data[row]
+
+    def rows(self) -> list:
+        return self.data
 
 
 class Table(Widget):
-    """ A Table Widget allows the disply of data in the from of columns and rows.
+    """ A Table Widget allows the display of data in the from of columns and rows.
 
     Args:
         headings (``list`` of ``str``): The list of headings for the table.
         id (str): An identifier for this widget.
+        data (``list`` of ``tuple``): The data to be displayed on the table.
         style (:class:`colosseum.CSSNode`): An optional style object.
-            If no style is provided then a new one will be created for the widget.
+            If no style is provided` then a new one will be created for the widget.
+        on_select(``callable``): A function to be invoked on selecting a row of the table.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
+
+    Examples:
+        >>> headings = ['Head 1', 'Head 2', 'Head 3']
+        >>> data = [('item 1', 'item 2', 'item3'),
+        >>>         ('item 1', 'item 2', 'item3')]
+        >>>
+        >>> table = Table(headings, data=data)
     """
+
     def __init__(self, headings, id=None, style=None, data=None, on_select=None, factory=None):
         super().__init__(id=id, style=style, factory=factory)
         self.headings = headings
-        self._data = data
+        self._data = None
         self._impl = self.factory.Table(interface=self)
+        self.data = data
 
         self.on_select = on_select
 
     @property
     def data(self):
-        '''
-        :returns: The data source of the tree
-        :rtype: ``dict``
-        '''
-        return self._data
+        """ The data source of the widget. It accepts table data
+        in the form of ``list``, ``tuple``, or :obj:`ListDataSource`
+
+        Returns:
+            (list) Returns a ``list`` of lists. Where the outer lists represents the
+            rows and each inner list represents a column.
+        """
+        return self._data if self._data is not None else None
 
     @data.setter
     def data(self, data):
-        '''
-        Set the data source of the data
-
-        :param data: Data source
-        :type  data: ``list``, ``tuple``, or ``class``
-        '''
         if isinstance(data, (list, tuple)):
             self._data = ListDataSource(data)
         else:
             self._data = data
 
         if data is not None:
-            self._data.interface = self
-
-        self._impl.refresh()
+            # self._data.interface = self
+            self._impl.refresh()
```

Comment:
Won't this be required even on the "data is None" case? If I clear the data source, won't I need to redraw the (now empty) widget?